nixos/modules/programs/gamescope.nix

```diff
@@ -0,0 +1,64 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.gamescope;
+
+  gamescope = pkgs.gamescope.override {
+    extraWrapperArgs = optional (cfg.args != [ ])
+      ''--add-flags "${toString cfg.args}"''
+    ++ builtins.attrValues (mapAttrs (var: val: "--set-default ${var} ${val}") cfg.env);
+  };
+in
+{
+  options.programs.gamescope = {
```

Comment:
Can you add a `package` option? That should allow for overriding the `gamescope` derivation this module uses more easily.